subprojects/base-services/src/main/java/org/gradle/util/internal/TextUtil.java

```diff
@@ -207,4 +207,34 @@ public static String camelToKebabCase(String camelCase) {
     public static String toLowerCaseLocaleSafe(String s) {
         return s.toLowerCase(Locale.ENGLISH);
     }
+
+    /**
+     * Replacement for locale-dependent {@code String.toLowerCase()}.
+     *
+     * <p>In most cases, making a string lowercase should be independent from the current locale (i.e. {@code String.toLowerCase(Locale.XXX)} should be used). If an implementation does need to be
+     * locale-dependent then this method has to be used.
```

Comment:
I'm not sure which method you're referring to here with "this method". It sort of sounds like you're saying to use toLowerCaseUserLocale if you want a version of toLowerCase that _does not_ use the user's locale. I think it's the opposite:

- toLowerCaseUserLocale needs to be used if the way we're using the String needs to be the user's locale.
- toLowerCaseLocaleSafe needs to be used if the way we're using the String needs to be a specific locale (English) regardless of the user's locale.

Reply:
I've rephrased the text.